thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Linker.java

```diff
@@ -292,42 +293,48 @@ private void register(Named type) {
 
     @Nonnull
     ThriftType resolveType(TypeElement type) {
+        AnnotationElement annotationElement = type.annotations();
+        Map<String, String> annotations = annotationElement != null
+                ? annotationElement.values()
+                : Collections.<String, String>emptyMap();
+
         ThriftType tt = typesByName.get(type.name());
         if (tt != null) {
-            return tt;
+            // If we are resolving e.g. the type of a field element, the type
+            // may carry annotations that are not part of the canonical type.
+            if (annotations.isEmpty()) {
```

Comment:
should be *not* `annotations.isEmpty()`.

Add some unit tests to verify, Travis shouldn't have let this through.